src/peer_connection.cpp

```diff
@@ -2124,7 +2124,7 @@ namespace libtorrent
 		if (t->share_mode()) return false;
 
 		if (m_upload_only && t->is_upload_only()
-			&& can_disconnect(error_code(errors::upload_upload_connection, get_libtorrent_category())))
+			&& can_disconnect(errors::upload_upload_connection))
```

Comment:
Question: If `error_code(` is not necessary (or maybe it's), is it a good idea to remove the rest?